### PR TITLE
Add `And` and `Or` steps

### PR DESCRIPTION
**Pull request by petrjanda**
Add `And` and `Or` steps according to http://tinkerpop.incubator.apache.org/docs/3.0.0.M9-incubating/#and-step

### CODE REVIEW

File: src/main/scala/gremlin/scala/GremlinScala.scala [hunk 1]
```diff
@@ -351,6 +351,9 @@ object GremlinScala {
 
     def hasNot(key: String, value: Any) = GremlinScala[End, Labels](traversal.where(P.not(__.has[End](key, value))))
 
+    def and(traversals: GremlinScala[End, HNil]*) =
+      GremlinScala[End, Labels](traversal.and(traversals.map(_.traversal):_*))
```
Comment (petrjanda) [inline src/main/scala/gremlin/scala/GremlinScala.scala]: Given I don't have much experience with `HNil` I wasn't sure about typing here, just tried to follow other steps. Could it be better perhaps?

File: src/test/scala/gremlin/scala/LogicalSpec.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package gremlin.scala
+
+class LogicalSpec extends TestBase {
+
+  describe("and steps") {
+
+    it("returns a vertex if both conditions are met") {
+      val x = v(1)
+
+      x.and(
```
Comment (petrjanda) [inline src/test/scala/gremlin/scala/LogicalSpec.scala]: Thinking more about this ... I guess syntax we want to rather achieve is:

```scala
x.and(
  _.out().has("name", "lop"),
  _.out().has("name", "vadas")
).values[String]("name").toSet should be(Set("marko"))
```

right? Would `and` annotation be `def and(traversals: (End ⇒ GremlinScala[End, HNil])*)`?

**Comment by mpollmeier** [comment]
Thanks Petr, nice one! 
Re your comments (which I interestingly can't see on github, but they came via mail): the second type parameter holds the labels, so in this case it doesn't matter if they are HNil or whatever else. Therefor I'll change it to `_` and merge it in.

**Comment by mpollmeier** [comment]
interestingly i had to revert my change.. don't have the time right now to investigate, but anyway all your changes are merged in. many thanks!

**Comment by mpollmeier** [comment]
got it sorted in d332955. thanks again!

**Pull request merged**